geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/MessageDispatcher.java

```diff
@@ -193,6 +194,15 @@ public boolean isWaitingForReAuthentication() {
     return waitForReAuthenticationStartTime > 0;
   }
 
+  private volatile boolean subjectUpdated = false;
```

Comment:
`volatile` is redundant since all access is performed under `synchronized` block.